internal/app/relayer/repostitory/ichain.go

```diff
@@ -0,0 +1,12 @@
+package repostitory
+
+type IChain interface {
+	GetBlockAndPackets(height uint64) (interface{}, error)
+	GetBlockHeader(height uint64) (interface{}, error)
+	GetLightClientState(chainName string) (interface{}, error)
+	GetLightClientConsensusState(chainName string, height uint64) (interface{}, error)
+	GetStatus() (interface{}, error)
+	GetLatestHeight() (uint64, error)
+	Delay() uint64
```

Comment:
```suggestion
	GetLightClientDelay() uint64
```
We don't know the self-delay of the chain, and can only get it from the light client of the counterparty chain.

Reply:
There are two delay configurations, delay block and delay time, which seem to be considered. cc @dreamer-zq